In [1]:
# Add the src directory to the Python path
import sys
import pandas as pd
sys.path.append('../src')
# Import necessary modules
from hypothesis_testing import HypothesisTester

In [2]:

# Load the processed data
cleaned_data = pd.read_csv('../data/Processed/cleaned_data.csv', low_memory=False)


In [3]:
tester = HypothesisTester(cleaned_data)

### Risk Differences Across Provinces
#### Analysis Objective
This test examines whether there are significant differences in risk levels (measured by Total Claims) across provinces. The goal is to understand how risk varies regionally, which can inform province-specific policies or risk management strategies.
### Hypothesis
Null Hypothesis (H₀): No risk differences across provinces.

Alternative Hypothesis (H₁): Risk differences exist across provinces.

In [8]:

# 1. Test risk differences across provinces
province_result = tester.test_risk_by_group('Province', 'TotalClaims')
print(province_result)

Test: ANOVA
Statistic: 5.849413762407606
P-value: 1.6782057588675903e-07
Message: Reject Null Hypothesis: There are significant differences.
None


#### Results

- Test Type: ANOVA (Analysis of Variance)
  - Compares the mean Total Claims across multiple provinces.
- F-Statistic: 5.849
  - Indicates significant variation in Total Claims between provinces.
- p-Value: 1.6782e-07
 - This value is much smaller than the common significance level of 0.05 indicating that the observed differences are statistically significant.
- Decision: Reject the null hypothesis (H₀).
#### Conclusion
There is sufficient evidence to conclude that significant risk differences exist across provinces. The variations in Total Claims are statistically significant and not due to random chance.
#### Implications
1. Targeted Risk Strategy: Risk management or pricing strategies may need to be adjusted by province.
2. Further Exploration: Identify specific provinces driving the differences and tailor policies accordingly.
3. Potential Policy Adjustments: Review how premiums and policies can be aligned with the provincial risk profiles.
#### Next Steps
Visualize Total Claims by province to pinpoint specific regional disparities.

Investigate other factors contributing to the risk differences between provinces.